webrender/src/device.rs

```diff
@@ -1532,36 +1534,67 @@ impl Device {
                              data);
     }
 
-    pub fn update_texture(&mut self,
-                          texture_id: TextureId,
-                          x0: u32,
-                          y0: u32,
-                          width: u32,
-                          height: u32,
-                          data: &[u8]) {
-        debug_assert!(self.inside_frame);
-
-        let mut expanded_data = Vec::new();
-
-        let (gl_format, bpp, data) = match self.textures.get(&texture_id).unwrap().format {
+    fn get_texture_info<'a>(&mut self, texture_id: TextureId, data: &'a [u8]) -> (u32, u32, Cow<'a, [u8]>) {
+        match self.textures.get(&texture_id).unwrap().format {
             ImageFormat::A8 => {
                 if cfg!(any(target_arch="arm", target_arch="aarch64")) {
+                    // Since we can allocate data here, i don't think we can
```

Comment:
I'm not sure why we need this - can we just have a single update_texture function which always expects stride? In which case there's no need for the get_texture_info() function, and thus no need for Cow.